### PR TITLE
Suppress compiler warning on Centos

### LINKED ISSUE #289: Centos Bullet version triggers compilation warnings
## Bug report

**Required Info:**

- Operating System:
  - CentOS CI (7)
- Installation type:
  - Source
- Version or commit hash:
  - master

#### Steps to reproduce issue

Nightly CentOS builds have an additional compiler warning that is turning the build yellow.

Example Job: https://ci.ros2.org/view/nightly/job/nightly_linux-centos_debug/480/

Ouput:

```
[ 83%] [32mBuilding CXX object CMakeFiles/test_bullet.dir/test/test_tf2_bullet.cpp.o[0m
In file included from [01m[K/usr/include/bullet/LinearMath/btVector3.h:21[m[K,
                 from [01m[K/usr/include/bullet/LinearMath/btMatrix3x3.h:19[m[K,
                 from [01m[K/usr/include/bullet/LinearMath/btTransform.h:21[m[K,
                 from [01m[K/home/jenkins-agent/workspace/nightly_linux-centos_debug/ws/src/ros2/geometry2/tf2_bullet/include/tf2_bullet/tf2_bullet.h:35[m[K,
                 from [01m[K/home/jenkins-agent/workspace/nightly_linux-centos_debug/ws/src/ros2/geometry2/tf2_bullet/test/test_tf2_bullet.cpp:32[m[K:
[01m[K/usr/include/bullet/LinearMath/btScalar.h:337:14:[m[K [01;35m[Kwarning: [m[K‘[01m[KbtInfinityMask[m[K’ defined but not used [[01;35m[K-Wunused-variable[m[K]
  static  int [01;35m[KbtInfinityMask[m[K = 0x7F800000;
              [01;35m[K^~~~~~~~~~~~~~[m[K
```

#### Additional information

Newer versions of bullet have an additional statement to prevent this compiler warning from happening: https://github.com/bulletphysics/bullet3/blob/7584847f334c03475665064d5217a5774a17dbfd/src/LinearMath/btScalar.h#L335-L338 but the version that is used on Centos (2.82) does not contain this fix.

It could be fixed with the workaround like this: https://pybullet.org/Bullet/phpBB3/viewtopic.php?t=9537

### PR DESCRIPTION
**Pull request by mjcarroll**
Closes #289 

Signed-off-by: Michael Carroll <michael@openrobotics.org>

### CODE REVIEW

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11621/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6755/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9503/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11587/)

* Centos: [Build Status](https://ci.ros2.org/job/ci_linux-centos/72/)

File: tf2_bullet/include/tf2_bullet/tf2_bullet.h [hunk 1]
```diff
@@ -32,12 +32,22 @@
...
+#include <LinearMath/btScalar.h>
 #include <LinearMath/btTransform.h>
 #include <geometry_msgs/msg/point_stamped.hpp>
 #include <tf2_ros/buffer_interface.h>
 
 #include <iostream>
 
+#if (BT_BULLET_VERSION == 282)
+// Suppress compilation warning on older versions of Bullet.
+// TODO(anyone): Remove this when all platforms have the fix upstream.
```
Comment (dirk-thomas) [inline tf2_bullet/include/tf2_bullet/tf2_bullet.h]: @mjcarroll Your name should go here.

File: tf2_bullet/include/tf2_bullet/tf2_bullet.h [hunk 2]
```diff
@@ -32,12 +32,22 @@
...
 
 #include <tf2/convert.h>
+#include <LinearMath/btScalar.h>
 #include <LinearMath/btTransform.h>
 #include <geometry_msgs/msg/point_stamped.hpp>
 #include <tf2_ros/buffer_interface.h>
 
 #include <iostream>
 
+#if (BT_BULLET_VERSION == 282)
```
Comment (dirk-thomas) [inline tf2_bullet/include/tf2_bullet/tf2_bullet.h]: Shouldn't the condition also be true for older versions?

Comment (mjcarroll) [reply tf2_bullet/include/tf2_bullet/tf2_bullet.h]: I didn't investigate older versions, as 2.82 is the oldest available on any platform that we support.

File: tf2_bullet/include/tf2_bullet/tf2_bullet.h (hunk 1 above)
Comment (mjcarroll) [reply tf2_bullet/include/tf2_bullet/tf2_bullet.h]: Done in f73cca9

File: tf2_bullet/include/tf2_bullet/tf2_bullet.h (hunk 2 above)
Comment (dirk-thomas) [reply tf2_bullet/include/tf2_bullet/tf2_bullet.h]: I should be easy to check if older versions also lack the function and to update the condition to match the comment.

Comment (mjcarroll) [reply tf2_bullet/include/tf2_bullet/tf2_bullet.h]: Older versions do lack the function.  2.82 is the current oldest release available on their official website, but I updated to `<=` for completeness.

**Review by dirk-thomas — APPROVED** [review]
LGTM with passing CI builds.

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11662/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6786/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9537/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11634/)
* Centos: [Build Status](https://ci.ros2.org/job/ci_linux-centos/73/)

**Comment by cottsay** [comment]
FYI, an update to Bullet in CentOS 7 was just submitted that should resolve these warnings in the system package: https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2020-8af9097046

Once the update hits the testing repository, I'll do a build to confirm.

**Pull request merged**